tag-system/src/main/java/org/togetherjava/tjbot/tag/TagSystem.java

```diff
@@ -0,0 +1,114 @@
+package org.togetherjava.tjbot.tag;
+
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.Tags;
+import org.togetherjava.tjbot.db.generated.tables.records.TagsRecord;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Tag system database utility.<br>
+ * Has methods to store & retrieve tags from the database.
+ *
+ * @author illuminator3
+ */
+public final class TagSystem {
+    private final Database database;
+
+    public TagSystem(Database database) {
+        this.database = database;
+    }
+
+    /**
+     * Checks if a tag already exists in the database
+     *
+     * @param tag id tag to check
+     * @return true if it exists, false if not
+     * @author illuminator3
+     */
+    public boolean exists(String tag) {
+        return database.readTransaction(ctx -> {
+            return Optional
```

Comment:
Do not abuse `Optional` as if-else. Return `... != null` instead.

Reply:
changed